src/common.rs

```diff
@@ -2,6 +2,29 @@
 
 use arrayvec::ArrayVec;
 
+/// Find real roots of a polynomial up to order of 3.
+/// Returns values of x for which c0 + c1 x + c2 x² + c3 x³ = 0, allowing for the coefficients
+/// to be equal to 0.
+pub(crate) fn solve_poly_3(c0: f64, c1: f64, c2: f64, c3: f64) -> ArrayVec<[f64; 3]> {
+    if c3 != 0. {
+        return solve_cubic(c0, c1, c2, c3);
+    };
+
+    let mut roots = ArrayVec::new();
+
+    if c2 != 0. {
```

Comment:
*Most* of this logic is redundant with checks already in `solve_quadratic`, but I see a special case to handle all coefficients 0. This is interesting, because all values of x satisfy the equation in this case. I'm wondering whether it would be better to move this logic to solve_quadratic.

In any case, returning a single "root" in the all-zero coefficient case should probably be added to the documentation.